node/full_node_integration_test.go

```diff
@@ -133,6 +133,158 @@ func TestTxGossipingAndAggregation(t *testing.T) {
 	}
 }
 
+func TestHeaderExchange(t *testing.T) {
+	testSingleAggreatorSingleFullNode(t)
+	testSingleAggreatorTwoFullNode(t)
+	testSingleAggreatorSingleFullNodeTrustedHash(t)
+	testSingleAggreatorSingleFullNodeSingleLightNode(t)
+}
+
+func testSingleAggreatorSingleFullNode(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	var wg sync.WaitGroup
+	aggCtx, aggCancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(context.Background())
+	clientNodes := 1
+	nodes, _ := createNodes(aggCtx, ctx, clientNodes+1, false, &wg, t)
+
+	node1 := nodes[0]
+	node2 := nodes[1]
+
+	require.NoError(node1.Start())
+	time.Sleep(2 * time.Second) // wait for more than 1 blocktime for syncer to work
```

Comment:
[question] Why do we need to wait here?

Reply:
node1 is sequencer and node2 is fullnode. for node2 to start the header exchange service, it needs the genesis block (or hash), so wait is introduced here such that genesis block becomes available and node2 can start gracefully.